pkg/models/datasource_cache.go

```diff
@@ -258,6 +308,10 @@ func (ds *DataSource) GetTLSConfig() (*tls.Config, error) {
 	return tlsConfig, nil
 }
 
+func (ds *DataSource) GetTLSConfig2(httpClientProvider httpclient.Provider) (*tls.Config, error) {
```

Comment:
This will be renamed before merge to `GetTLSConfig` and replace the existing `GetTLSConfig`